src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -1459,11 +1467,35 @@ void DatabaseCatalog::dropTableFinally(const TableMarkedAsDropped & table)
         table.table->drop();
     }
 
+    /// Check if we are interested in a particular disk
+    ///   or it is better to bypass it e.g. to avoid interactions with a remote storage
+    auto look_on_disk =
+        [this](DiskPtr disk, std::shared_ptr<MergeTreeData> tbl)
+    {
+        if (!tbl)
+            /// Not a MergeTree - don't skip
+            return true;
+
+        if (tbl->getStoragePolicy()->tryGetVolumeIndexByDiskName(disk->getName()).has_value())
+            /// Disk is actually used - don't skip
+            return true;
+
+        auto is_local_metadata = [&]() { return !(disk->isRemote() && disk->isPlain()); };
```

Comment:
Let's just use `isRemote`

Reply:
Are you sure? Why?
And remove '... that have local metadata"?

Reply:
The `isPlain()  == false` API does not guarantee that the disk will be local (i.e. `MetadataStorageType::Keeper`)